generic_config_updater/gu_common.py

```diff
@@ -113,18 +113,43 @@ def validate_sonic_yang_config(self, sonic_yang_as_json):
     def validate_config_db_config(self, config_db_as_json):
         sy = self.create_sonic_yang_with_loaded_models()
 
+        # TODO: Move these validators to YANG models
+        supplemental_yang_validators = [self.validate_bgp_peer_group,
+                                        self.validate_unique_lanes]
+
         try:
             tmp_config_db_as_json = copy.deepcopy(config_db_as_json)
 
             sy.loadData(tmp_config_db_as_json)
 
             sy.validate_data_tree()
 
-            # TODO: modularize custom validations better or move directly to sonic-yang module
-            return self.validate_bgp_peer_group(config_db_as_json)
+            for supplemental_yang_validator in supplemental_yang_validators:
+                success, error = supplemental_yang_validator(config_db_as_json)
+                if not success:
+                    return success, error
         except sonic_yang.SonicYangException as ex:
             return False, ex
 
+        return True, None
+
+    def validate_unique_lanes(self, config_db):
+        if "PORT" not in config_db:
+            return True, None
+
+        ports = config_db["PORT"]
+        existing = {}
+        for port in ports:
+            attrs = ports[port]
+            if "lanes" in attrs:
+                lanes_str = attrs["lanes"]
+                lanes = lanes_str.split(", ")
+                for lane in lanes:
+                    if lane in existing:
+                        return False, f"'{lane}' lane is used multiple times in PORT: {set([port, existing[lane]])}"
+                    existing[lane] = port
+        return True, None
```

Comment:
>None [](http://example.com/codeflow?start=21&length=4)

return `""` instead of `None`.

Reply:
In C#, we use the pattern `TryParse(...)` the return is a boolean and the output is passed as an out paramter.

Here I think I am doing the same, but instead `out param`, returning 2 outputs

In C#, we return `null` as the result of parsing if parsing fails. This is how the built in class `IPAddress` handles `TryParse`

```csharp
IPAddress.TryParse("invalid", out IPAddress invalid_ip);
Console.WriteLine(invalid_ip);
```
Output
```
null
```

Is that not good in python? or you want this improvement for reduce the four-letter word `None` to `""`

Reply:
It's a minor issue, not critical.

Reply:
Ah, did not know we can do that with assert. I think for the current case if the return is `False` the error should neither be `None` or `Empty`. I think this is interesting let me think about it separately.